src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -160,8 +148,17 @@ private bool RemoveSession(int sessionId, out ProfilerSession session)
         {
             lock (this.sessionsLock)
             {
+                //cancel running XEventStream for session.
+                CancellationTokenSource targetToken;
+                if (monitoredCancellationTokenSources.Remove(sessionId, out targetToken))
+                {
+                    targetToken.Cancel();
+                }
                 if (this.monitoredSessions.Remove(sessionId, out session))
                 {
+                    // Toggle isStreaming status of removed session to not streaming.
```

Comment:
This isn't really toggling (toggle implies switching to whatever it currently isn't set to)

But I don't think this needs a comment, it's clear enough from the name what's going on.